modules/@apostrophecms/schema/ui/apos/components/AposAttachment.vue

```diff
@@ -50,19 +56,78 @@ export default {
     },
     disabled () {
       return (this.limitReached || this.field.disabled);
+    },
+    moduleOptions() {
+      return window.apos.modules['@apostrophecms/file'];
     }
   },
   methods: {
     updated(items) {
       this.next = items;
     },
     validate(value) {
-      if (this.field.required && !value.data.length) {
+      if (this.field.required && !value) {
         return 'required';
       }
 
       return false;
-    }
+    },
+    async uploadMedia (event) {
+      this.$emit('upload-started');
+      const file = event.target.files[0];
+
+      const emptyDoc = await apos.http.post(this.moduleOptions.action, {
+        body: {
+          _newInstance: true
+        }
+      });
+      await apos.notify('Uploading file', { dismiss: true });
+
+      const formData = new window.FormData();
+
+      formData.append('file', file);
+      let attachment;
+
+      try {
+        attachment = await apos.http.post('/api/v1/@apostrophecms/attachment/upload', {
+          body: formData
+        });
+      } catch (error) {
+        console.error('Error uploading file.', error);
+
+        const msg = error.body && error.body.message ? error.body.message : 'Upload error';
+
+        await apos.notify(msg, {
+          type: 'danger',
+          icon: 'alert-circle-icon',
+          dismiss: true
+        });
+        return;
+      }
+
+      const fileData = Object.assign(emptyDoc, {
+        title: attachment.title,
+        attachment
+      });
+
+      try {
+        const uploaded = await apos.http.post(this.moduleOptions.action, {
+          body: fileData
+        });
+        await apos.notify('Upload Successful', {
+          type: 'success',
+          dismiss: true
+        });
+        this.value.data = {
+          ...uploaded[this.field.name],
```

Comment:
We are not making a new piece or page here. We are just editing a field. The field happens to be an attachment, so we get a new value for it by uploading a file, but otherwise it is like any other field. What is `uploaded` doing and why would the field ever need to look inside it for its own field name?

As I understand it the special case of creating an entire piece via an upload in the media manager has already been implemented elsewhere and doesn't impact this field.

A piece might have many attachment fields, they don't all create new pieces when they are edited.